File: 알고리즘 문제풀이/프로그래머스/모의고사.py
#1, 2, 3, 4, 5
#2, 1, 2, 3, 2, 4, 2, 5
#3, 3, 1, 1, 2, 2, 4, 4, 5, 5
def solution(answers):
    answer,check,checked,m = [],[[1,2,3,4,5],[2, 1, 2, 3, 2, 4, 2, 5],[3, 3, 1, 1, 2, 2, 4, 4, 5, 5]],[[],[],[]],0
    for i in range(3):
        checked[i] = sum([1 if answers[j]==check[i][j%len(check[i])]  else 0  for j in range(len(answers))])
    m = max(checked)
    for i in range(3):
        if checked[i]==m:
            answer.append(i+1)
    return answer

File: 알고리즘 문제풀이/프로그래머스/모의고사.js
function sum(arr) {
    return arr.reduce((x,y)=>x+y,0);
}

function solution(answers) {
    const answer = [];
    const userAnswers = [[1,2,3,4,5],[2,1,2,3,2,4,2,5],[3,3,1,1,2,2,4,4,5,5]];
    const checked = [[],[],[]];
    let maxCorrect = 0;
    for(let i=0;i<3;i++){
        for(let j=0;j<answers.length;j++){
            checked[i].push(userAnswers[i][j%userAnswers[i].length]===answers[j]?1:0)
        }
        maxCorrect = Math.max(maxCorrect,sum(checked[i]));
    }
    for(let i=0;i<3;i++){
        sum(checked[i])===maxCorrect&&answer.push(i+1);
    }
    
    return answer;
}